src/ai/backend/manager/api/session.py

```diff
@@ -2178,6 +2180,111 @@ async def shutdown_service(request: web.Request, params: Any) -> web.Response:
     return web.Response(status=204)
 
 
+async def find_dependency_sessions(
```

Comment:
In the test case of this issue, this function with t5 session id at first finds t4_1, t4_2, t4_3, since t4_1, t4_2, t4_3 are required to execute t5.
So, I think maybe t4_1, t4_2, and t4_3 are dependency of t5 session,
and t5 is a dependent session of t4_1, t4_2, and t4_3.